src/psyclone/gocean1p0.py

```diff
@@ -985,7 +989,9 @@ def __str__(self):
 
         # Generate the upper and lower loop bounds
         self.start_expr = self.lower_bound()
```

Comment:
I know it looks a bit verbose at the moment but whenever possible I would like to keep the parent setter in the line before or after the children is set,  in this case it is the `self.start_expr =`. It has been quite confusing in his PR when the two operations are done in different places.

Note that I want to attempt #294 next that would remove the `.parent =` expressions anyway.

Maybe you meant that` lower_bound()`/`upper_bound()` should set both the children and the parent connection. This may be better but again I would like to explore this separately as my preferred path would be to initialize the start/stop expressions at the `__init__` and let transformations (or setting a new iteration space) act on them.

Reply:
OK, just so long as you have it in hand then I'm happy.